rest_server/SiodbLoggerPool.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	FATAL_INIT_ERROR                   = 2
+	FATAL_UNABLE_TO_CLEANUP_TCP_BUFFER = 2
+)
+
+type SiodbLoggerPool struct {
+	siodbLogger []*SiodbLogger
+}
+
+func (loggerPool *SiodbLoggerPool) ConfigGinLogger() {
+
+	gin.SetMode(gin.ReleaseMode)
+	gin.DisableConsoleColor()
+	gin.DefaultWriter = ioutil.Discard
+
+}
+
+func (loggerPool *SiodbLoggerPool) LogRequest(c *gin.Context, latency time.Duration) {
+
+	path := c.Request.URL.Path
+	raw := c.Request.URL.RawQuery
+	if raw != "" {
+		path = path + "?" + raw
+	}
+
+	loggerPool.Output(INFO, "Request | %3d | %13v | %15s | %15v | %s | %-7s | %s",
+		c.Writer.Status(),
+		latency,
+		c.ClientIP(),
+		c.Request.UserAgent(),
+		c.Request.Method,
+		c.Request.Proto,
+		path,
+	)
+}
+
+func (loggerPool *SiodbLoggerPool) Output(logLevel int, s string, v ...interface{}) error {
+
+	for _, siodbLogger := range loggerPool.siodbLogger {
+		siodbLogger.Output(logLevel, s, v...)
+	}
+
+	return nil
+
+}
+
+func (loggerPool *SiodbLoggerPool) Trace(s string, v ...interface{}) {
+	loggerPool.Output(TRACE, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Debug(s string, v ...interface{}) {
+	loggerPool.Output(DEBUG, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Info(s string, v ...interface{}) {
+	loggerPool.Output(INFO, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Warning(s string, v ...interface{}) {
+	loggerPool.Output(WARNING, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Error(s string, v ...interface{}) {
+	loggerPool.Output(ERROR, s, v...)
+}
+
+func (loggerPool *SiodbLoggerPool) Fatal(code int, s string, v ...interface{}) {
+	loggerPool.Output(FATAL, s, v...)
+	os.Exit(code)
+}
+
+func FormattedOutput(logLevel int, s string, v ...interface{}) string {
+
```

Comment:
extra empty line. might look somehow reasonable (but still quetionable) with functions having lots of lines of code, but in function with 2-3 lines of code no sense, absolutely